### ISSUE TITLE
After training, the parameter does not become 0, what should i do?

### ISSUE THREAD

**Issue by panchengl** [Pokemon-Huang/sparse-structure-selection-PyTorch#1]
hello, thanks your work,  i use ypur demo to train, but after training, the parameter does not become zero,
![image](https://user-images.githubusercontent.com/56785763/103132865-f19b3a00-46e1-11eb-9e06-237f5af7e5cb.png)

**Comment by panchengl** [comment]
lambda 'Parameter containing:
tensor([0.6660, 0.4774, 0.5917, 0.8245, 0.9268, 2.6580], device='cuda:0',
       requires_grad=True)'

**Comment by panchengl** [comment]
@Pokemon-Huang   Looking forward to your reply， thank you

**Comment by Pokemon-Huang** [comment]
@panchengl Are u running the algorithm on ResNet-20? I did remember this work can not prune this model, may be it's too shallow, try ResNet50.

**Comment by panchengl** [comment]
thanks your reply, yes, i use resnet-20, may i use resnet56 in  ciafar10 or cifar 100?  or resnet50 in imagenet?

**Comment by Pokemon-Huang** [comment]
resnet56 in cifar10 is ok, i didn't try the other either. Generally speaking, the algorithm works better with easier datasets and deeper models.

**Comment by panchengl** [comment]
ok, thank u , i want use blocks prune in object detections,  Do you think this algorithm is effective?

**Comment by Pokemon-Huang** [comment]
emmm, the unique optimization method of this algorithm will influence the accuracy, so it will work if your datasets is easy. BTW, this work publics in 2017, u can check some recent papers, like nas.

**Comment by panchengl** [comment]
Ok, thanks for your advices, i will try in my datasets, if work, i will reply to you ,&nbsp; thank u

---Original---

**Comment by Ta-SIR** [comment]
hello, thanks your work, why is there no R1 loss in main，but in sss
